src/core/util/byte_stream.cc

```diff
@@ -158,5 +158,20 @@ const std::string GetFormattedHex(const std::uint8_t* data, std::size_t size) {
   return hex.str();
 }
 
+std::unique_ptr<std::uint8_t[]> AddressToByteArray(
+    const boost::asio::ip::address& address,
+    std::size_t* size)
+{
+  *size = address.is_v4() ? 4 : 16;
+  std::unique_ptr<std::uint8_t[]> buffer =
+      std::make_unique<std::uint8_t[]>(*size);
+  std::memcpy(
+      buffer.get(),
+      address.is_v4() ? address.to_v4().to_bytes().data()
+                      : address.to_v6().to_bytes().data(),
+      *size);
+  return buffer;
+}
+
```

Comment:
>std::unique_ptr<<std::uint8_t[]>>
>std::size_t* size)

At this point, why not refactor/implement/return a vector? Simply pass the address object, do the checks, create vector from `to_bytes().data()` (or `assign` if passing arg reference to *caller* vector object), and return the vector (*if* creating the object locally in the *callee*)? From there, `Insert` the vector data pointer and vector size. We could probably implement all of that in just a few lines (including the return) while reducing the session code by a couple of lines.